File: Rotate image/Rotate_image.java
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        
        for (int i = 0; i < n; ++i) {
            reverseRow(matrix[i]);
        }
    }
    
    private void reverseRow(int[] row) {
        int m = row.length;
        for (int j = 0; j < m / 2; ++j) {
            int temp = row[j];
            row[j] = row[m - j - 1];
            row[m - j - 1] = temp;
        }
    }
}

File: Rotate image/Rotate_image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n = len(matrix)
    
        for i in range(n):
            for j in range(i + 1, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
                
        for i in matrix : 
            m = len(i)
            for j in range(m//2):
                i[j],i[m-j-1]=i[m-j-1],i[j]
        